server/routeHandlers.go

```diff
@@ -53,103 +53,107 @@ func newGame(c echo.Context) error {
+
+	if joinErr != nil {
 		return joinErr
 	}
-    
-    return respondWithJWTIfValid(c, player, joinErr == nil)
+
+	return respondWithJWTIfValid(c, player, joinErr == nil, isHost)
 }
 
 func startGame(c echo.Context) error {
-    
-    authHeader := c.Request().Header.Get(echo.HeaderAuthorization)
-    
-    claims, validUser := getValidClaimsFromHeader(authHeader)
+	authHeader := c.Request().Header.Get(echo.HeaderAuthorization)
+
+	claims, validUser := getValidClaimsFromHeader(authHeader)
 
 	if !validUser || claims["isHost"] != true {
 		return c.JSONPretty(http.StatusUnauthorized, &Response{false, nil}, " ")
 	}
-    
+
 	dealCards()
 	return update(c)
 }
 
 func update(c echo.Context) error {
-    
-    authHeader := c.Request().Header.Get(echo.HeaderAuthorization)
-    
-    claims, validUser := getValidClaimsFromHeader(authHeader)
+
+	authHeader := c.Request().Header.Get(echo.HeaderAuthorization)
+
+	claims, validUser := getValidClaimsFromHeader(authHeader)
 
 	if !validUser {
 		return c.JSONPretty(http.StatusUnauthorized, &Response{false, nil}, " ")
 	}
 
 	valid := updateGame(claims["gameid"].(string))
-	return respondIfValid(c, valid && validUser, claims["userid"].(string), claims["gameid"].(string))
+	return respondIfValid(c, valid && validUser, claims["name"].(string), claims["gameid"].(string))
 }
 
 func play(c echo.Context) error {
-    
-    authHeader := c.Request().Header.Get(echo.HeaderAuthorization)
-    
-    claims, validUser := getValidClaimsFromHeader(authHeader)
+	authHeader := c.Request().Header.Get(echo.HeaderAuthorization)
+
+	claims, validUser := getValidClaimsFromHeader(authHeader)
 
 	if !validUser {
 		return c.JSONPretty(http.StatusUnauthorized, &Response{false, nil}, " ")
 	}
-    
-    // TODO Cards have a value, which can include skip, reverse, etc
-    card := model.Card{c.Param("number"), c.Param("color")}
-	valid := playCard(claims["gameid"].(string), claims["userid"].(string), card)
-	return respondIfValid(c, valid, claims["userid"].(string), claims["gameid"].(string))
+
+	// TODO Cards have a value, which can include skip, reverse, etc
+	card := model.Card{c.Param("number"), c.Param("color")}
+	valid := playCard(claims["gameid"].(string), claims["name"].(string), card)
+	return respondIfValid(c, valid, claims["name"].(string), claims["gameid"].(string))
 }
 
 func draw(c echo.Context) error {
-    
-    authHeader := c.Request().Header.Get(echo.HeaderAuthorization)
-    
-    claims, validUser := getValidClaimsFromHeader(authHeader)
+	authHeader := c.Request().Header.Get(echo.HeaderAuthorization)
+	claims, validUser := getValidClaimsFromHeader(authHeader)
 
 	if !validUser {
 		return c.JSONPretty(http.StatusUnauthorized, &Response{false, nil}, " ")
 	}
 
-	valid := drawCard(claims["gameid"].(string), claims["userid"].(string))
-	return respondIfValid(c, valid, claims["userid"].(string), claims["gameid"].(string))
+	valid := drawCard(claims["gameid"].(string), claims["name"].(string))
```

Comment:
This would make it so that people who have the same name get mixed up (i.e. if our game had two people with the name "John"). We need to use the userid, as it is the only value guaranteed to be unique to the player.

Reply:
Perhaps we pass in the whole JWT token?

Reply:
I was using the "userid" but the uno server code labeled the parameter as "username". Perhaps I should have left it as "userid"

Reply:
The JWT token seems more like a route concept to me but if you see how it would be useful to the back-end I'm not opposed to that.

Reply:
My main concern here is if we pass only the name to the drawcard function, then which hand will uno add the card to? The first person named John or the second person named John? It needs a way to differentiate them. I was thinking that the userid would be a good way to do this.

However, I didn't look enough into how the name is being used. I thought it would be fine to switch it out for the userid, from what I could see. I could totally be wrong! Perhaps we need to pass in both the userid and the name.